client/tests/integration/smartcontracts/executor_with_custom_parameter/src/lib.rs

```diff
@@ -0,0 +1,68 @@
+//! Runtime Executor which allows domains whose id satisfies the length limit
+#![no_std]
+
+extern crate alloc;
+#[cfg(not(test))]
+extern crate panic_halt;
+
+use iroha_executor::{prelude::*, DataModelBuilder};
+use lol_alloc::{FreeListAllocator, LockedAllocator};
+
+#[global_allocator]
+static ALLOC: LockedAllocator<FreeListAllocator> = LockedAllocator::new(FreeListAllocator::new());
+
+getrandom::register_custom_getrandom!(iroha_executor::stub_getrandom);
+
+use alloc::format;
+
+mod parameter {
+    //! Module with custom parameters
+    use alloc::string::String;
+
+    use iroha_schema::IntoSchema;
+    use parity_scale_codec::{Decode, Encode};
+    use serde::{Deserialize, Serialize};
+
+    use super::*;
+
+    /// Parameter that controls domain limits
+    #[derive(PartialEq, Eq, Parameter, Decode, Encode, Serialize, Deserialize, IntoSchema)]
+    pub struct DomainLimits {
+        /// Length of domain id in bytes
+        pub id_len: u32,
+    }
+}
+
+#[derive(Constructor, ValidateEntrypoints, Validate, Visit)]
+#[visit(custom(visit_register_domain))]
+struct Executor {
+    verdict: Result,
+    block_height: u64,
+}
+
+fn visit_register_domain(executor: &mut Executor, _authority: &AccountId, isi: &Register<Domain>) {
+    let parameters = FindAllParameters.execute().unwrap().into_inner();
```

Comment:
Would be cool to have ability to filter params to get certain parameter value to decrease overhead. 
Just a thought certainly not for this PR.